tslearn/preprocessing.py

```diff
@@ -145,34 +149,40 @@ def transform(self, X, y=None, **kwargs):
         numpy.ndarray
             Rescaled time series dataset.
         """
-        if self.min_ is not None:
+        value_range = self.value_range
+        if self.min is not None:
             warnings.warn(
                 "'min' is deprecated in version 0.2 and will be "
                 "removed in 0.4. Use value_range instead.",
                 DeprecationWarning, stacklevel=2)
-            self.value_range = (self.min_, self.value_range[1])
+            value_range = (self.min, value_range[1])
 
-        if self.max_ is not None:
+        if self.max is not None:
             warnings.warn(
                 "'max' is deprecated in version 0.2 and will be "
                 "removed in 0.4. Use value_range instead.",
                 DeprecationWarning, stacklevel=2)
-            self.value_range = (self.value_range[0], self.max_)
+            value_range = (value_range[0], self.max)
 
-        if self.value_range[0] >= self.value_range[1]:
+        if value_range[0] >= value_range[1]:
             raise ValueError("Minimum of desired range must be smaller"
-                             " than maximum. Got %s." % str(self.value_range))
+                             " than maximum. Got %s." % str(value_range))
 
+        X = check_array(X, allow_nd=True, force_all_finite=False)
+        X = check_dims(X, self._X_fit, extend=False)
         X_ = to_time_series_dataset(X)
         min_t = numpy.min(X_, axis=1)[:, numpy.newaxis, :]
```

Comment:
If we allow nans (cf tags below), shouldn't we use https://docs.scipy.org/doc/numpy/reference/generated/numpy.nanmin.html ?

Reply:
Good call (and indeed same for nanmax, nanmean, nanstd, ...).

We should probably have tests for this in place as well.